### PR TITLE
fix #22: Update notification bar styling and behavior to match updated design spec

### LINKED ISSUE #22: Update notification bar styling and behavior to match updated design spec
## Design:
https://www.dropbox.com/s/nbkkatjfhhr2xbj/sk3.png?dl=0

blok-notification-bar-style-behavior

form controls assets are here:
http://people.mozilla.org/~shorlander/styleguide/styleguide-buttons/form-controls.html

## Copy:
[blokr-copy-reference](https://docs.google.com/presentation/d/1_O-A3cKMS4cto51Ld5uiyEmXRZ1HxGL_RmHQNzR0atg/edit?ts=578e5262#slide=id.p)

https://docs.google.com/spreadsheets/d/19JAW3XkhfkRbAeGKia3loRMZou96_HUVoBxSh3f1oEQ/edit?ts=578e51f8#gid=0

### PR DESCRIPTION
**Pull request by groovecoder**
blok-cnn-toolbar

blok-cnn-disabled

blok-cnn-toolbar-page-works

blok-cnn-feedback-modal

blok-cnn-toolbar-broken-feedback

### CODE REVIEW

**Comment by groovecoder** [comment]
@rpl - can I get some more JS review here? (Keeping in mind I've still got https://github.com/mozilla/blok/issues/24, https://github.com/mozilla/blok/issues/25, and https://github.com/mozilla/blok/issues/26 already filed for follow-up enhancements and re-factorings.)

**Comment by rpl** [comment]
@groovecoder sure, I'll look at it asap

File: js/contentscript.js [hunk 1]
```diff
@@ -1,15 +1,11 @@
-var toolbarFrame;
+var toolbarFrame, feedbackOverlay;
 
 if (window.parent == window) {
```
Comment (rpl) [inline js/contentscript.js]: @groovecoder if a content script registered in the manifest only needs to be injected in the top level frame, then we should take into account that there is a [manifest property named "all_frames"](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/content_scripts), that defaults to `false` and should be enough to filter out the nested iframes without doing the necessary checks (e.g. with the above `window.parent == window` check) manually.

is there any issue/bug in the "all_frames" property behavior that needed to be fixed with this manual check?

Comment (groovecoder) [inline js/contentscript.js]: No issue or bug; I simply started with the code from https://github.com/mdn/webextensions-examples/blob/200c7a35999629c3095ef4f82f07fd0c09116235/inpage-toolbar-ui/contentscript.js.

I sent https://github.com/mdn/webextensions-examples/pull/79 upstream so others don't make a similar mistake.

**Comment by rpl** [comment]
I understand that the reason of these UI changes is to match the new design described in #22, but I have to point out that there are issues with injecting iframes into arbitrary webpages that have to be taken into account (at least to be sure if they matter or not in the particular scenario that an add-on has as its target), more details about this have been added into the following bugzilla issue:

https://bugzilla.mozilla.org/show_bug.cgi?id=1287590

Long story short, we have to take into account that the webpage cannot use the injected iframe to access the WebExtensions APIs, but it could easily use the injected iframe to detect that the addon is injecting the iframe and react in some way, and page can even remove or tweak the style of the iframe (without the need to access its content).

In this particular scenario (ad-blocking) this issue is probably not secondary.

File: js/toolbar.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
 browser.runtime.onMessage.addListener(function (runtimeMessage) {
-  var blockedCount, blockedForm;
+  var blockedCount, blockedForm, allowedCount;
```
Comment (rpl) [inline js/toolbar.js]: @groovecoder `blockedForm` seems to be currently unused, it is probably a leftover from a previous version of this toolbar.

File: js/toolbar.js [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
   if (runtimeMessage.hasOwnProperty('origin-disabled')) {
+    allowedCount = runtimeMessage.allowed_requests.length;
+
     document.querySelector('#title-blocking').className = 'hide';
     document.querySelector('#title-disabled').className = 'title';
-    document.querySelector('#title-origin').innerHTML = runtimeMessage['origin-disabled'];
-    document.querySelector('#disable').className = 'hide';
+    document.querySelector('#title-allowed-count').innerHTML = allowedCount;
+
+    for (feedbackElement of document.querySelectorAll('.feedback')) {
```
Comment (rpl) [inline js/toolbar.js]: @groovecoder `feebackElement` is not explicitly defined (I mean as in `for (let feedbackElement of ...) { ... }`).

File: js/toolbar.js [hunk 4]
```diff
@@ -16,24 +22,21 @@ browser.runtime.onMessage.addListener(function (runtimeMessage) {
...
-    browser.runtime.sendMessage("close-toolbar");
-  });
-}
-
-for (reasonBtn of document.querySelectorAll('.reason')) {
-  reasonBtn.addEventListener('click', function (event) {
-    browser.runtime.sendMessage({"disable-reason": event.target.text});
-    document.querySelector('#disable-reasons').className = 'hide';
-    document.querySelector('#disable-reason-thankyou').className = '';
+for (feedbackBtn of document.querySelectorAll('.feedback-btn')) {
```
Comment (rpl) [inline js/toolbar.js]: @groovecoder define `feedbackBtn` explicitly (as for `feedbackElement` in the comment above)

File: js/toolbar.js [hunk 5]
```diff
@@ -16,24 +22,21 @@ browser.runtime.onMessage.addListener(function (runtimeMessage) {
...
+  feedbackBtn.addEventListener('click', function (event) {
+    var feedback = event.target.dataset.feedback;
+    browser.runtime.sendMessage({"feedback": feedback});
+    for (feedbackDiv of document.querySelectorAll('.feedback')) {
```
Comment (rpl) [inline js/toolbar.js]: @groovecoder same for `feedbackDiv`.

File: js/toolbar.js [hunk 6]
```diff
@@ -16,24 +22,21 @@ browser.runtime.onMessage.addListener(function (runtimeMessage) {
   }
 });
 
-document.querySelector('#disable-btn').addEventListener('click', function (event) {
+document.querySelector('#disable-link').addEventListener('click', function (event) {
```
Comment (rpl) [inline js/toolbar.js]: @groovecoder nit, the `event` parameter is not used in this callback and can be omitted (same on line 29)

File: js/contentscript.js [hunk 7]
```diff
@@ -1,15 +1,11 @@
-var toolbarFrame;
+var toolbarFrame, feedbackOverlay;
```
Comment (rpl) [inline js/contentscript.js]: @groovecoder this should probably be `feedbackModalOverlay` instead of `feedbackOverlay`.

File: js/feedback.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+for (closeBtn of document.querySelectorAll('.close-feedback')) {
```
Comment (rpl) [inline js/feedback.js]: @groovecoder explicitly declare `closeBtn` in this `for...of`.

File: js/feedback.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  closeBtn.addEventListener('click', function (event) {
```
Comment (rpl) [inline js/feedback.js]: nit, event is currently unused in this listener (and in the listener at line 7 as well)

**Comment by rpl** [comment]
@groovecoder thanks for your patience, I would wanted to complete it yesterday, my apologies for the waiting time.

From a JS review point of view, I think that now we've collected the most important pieces to highlight.

As another side-note, I noticed that we have a number of `console.log` in the background, the content script and the extension iframes scripts, it would be nice to add a follow up issue to make this console statements something that happens only when the extension has been built (or to be more precise, "browserified") in development mode, and to omit the logs in the production build (and if it needs to be logged in a production build, then it is probably something that we would want to turn into a `console.warn` or a `console.error`).

**Comment by groovecoder** [comment]
Thanks @rpl. Do you have a tool suggestion for stripping out the `console.log` statements?

@tatumcreative pointed me at https://github.com/alanshaw/stripify and/or https://github.com/hughsk/envify

**Comment by rpl** [comment]
@groovecoder I was thinking exactly to what @tatumcreative suggests! :-) (cut them out as part of the bundling)

I never tried stripify but it seems to do exactly what we want currently.

I used envify from time to time: it doesn't cut the code from the bundle, but it is a flexible solution if we want to conditionally execute a block of code by build type.

**Comment by groovecoder** [comment]
[stripify doesn't work with ES6](https://github.com/alanshaw/stripify/issues/8). 😢

`envify` seems like I'll have to wrap all `.log` statements in conditionals (gross), or shim the `console` object? Hmm ... do we think this should block merge? Or file as a follow-up issue?

**Comment by rpl** [comment]
@groovecoder probably the easiest way to use envify to wrap the `console.*` statements only once is to move them into a "logging" commonjs module which is the only one that uses envify to turn them into nop when it is a production build.

If I'm not wrong this add-on is still not officially released, in that case I don't see anything wrong to defer this part to a separate follow-up issue (and it makes even more sense given that it isn't the goal of this PR, and it would require to turn the other into browserify bundles all the javascript sources).

The upside of it is that once we start to browserify the other javascript sources, we can start to cover much more of them with unit tests.

**Comment by groovecoder** [comment]
Okay, https://github.com/mozilla/blok/pull/29/commits/ee22db7d8cc55fd9bbf56693b60a4cfcc64f2427 got `envify` to no-op log messages while bundling for the `.xpi`.

**Comment by rpl** [comment]
@groovecoder That's great! how about defining a new script in the package.json to build a development version? (e.g. with `npm run build-dev`)

**Comment by groovecoder** [comment]
https://github.com/mozilla/blok/pull/29/commits/f6174b7a91aafc495d303977be7a8afb37003c2c shows the number of tracked *entities* - not the number of tracked requests.

Note to self: need to plan a 1-2 day refactor of the global state in background.js into Redux.

**Comment by groovecoder** [comment]
Alright, the latest commit on this PR should address everything. I've merged it so that master will be the latest and I can fix https://github.com/mozilla/blok/issues/30 to see a full end-to-end install-Blok-via-Test-Pilot test.
